src/components/date/date.pw.tsx

```diff
@@ -0,0 +1,1349 @@
+      }
+    });
+  });
+
+  ([
+    [SIZE.SMALL, 30],
+    [SIZE.MEDIUM, 38],
+    [SIZE.LARGE, 46],
+  ] as [DateInputProps["size"], number][]).forEach(([size, approxValue]) => {
+    test(`should check the ${size} of the DateInput`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<DateInputCustom size={size} />);
+
+      const inputParent = getDataElementByValue(page, "input").locator("..");
+      await inputParent.click();
+      await inputParent.waitFor();
+      await assertCssValueIsApproximately(inputParent, "height", approxValue);
+    });
+  });
+
+  [
+    [10, 90, 135, 1229],
+    [30, 70, 409, 956],
+    [80, 20, 1092, 273],
+  ].forEach(([label, input, labelRatio, inputRatio]) => {
+    test(`should use ${label} as labelWidth, ${input} as inputWidth and render it with correct label and input width ratios`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <DateInputCustom labelInline labelWidth={label} inputWidth={input} />
+      );
+
+      const labelElement = getDataElementByValue(page, "label").locator("..");
+      await assertCssValueIsApproximately(labelElement, "width", labelRatio);
+      const inputElement = getDataElementByValue(page, "input").locator("..");
+      await assertCssValueIsApproximately(inputElement, "width", inputRatio);
+    });
+  });
+
+  ["10%", "30%", "50%", "80%", "100%"].forEach((maxWidth) => {
+    test(`should check maxWidth as ${maxWidth} for Date Input component`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<DateInputCustom maxWidth={maxWidth} />);
+
+      const inputParent = getDataElementByValue(page, "input")
+        .locator("..")
+        .locator("..");
+      await expect(inputParent).toHaveCSS("max-width", maxWidth);
+    });
+  });
+
+  test(`when maxWidth has no value it should render as 100%`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DateInputCustom inputWidth={100} maxWidth="" />);
+
+    const inputParent = getDataElementByValue(page, "input")
+      .locator("..")
+      .locator("..");
+    await expect(inputParent).toHaveCSS("max-width", "100%");
+  });
+
+  test(`should render Date with disabled prop`, async ({ mount, page }) => {
+    await mount(<DateInputCustom disabled />);
+
+    const input = getDataElementByValue(page, "input");
+    await expect(input).toBeDisabled();
+  });
+
+  test(`should render Date icon with disabled style`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DateInputCustom disabled />);
+
+    const calendarIcon = getDataElementByValue(page, "calendar");
+    await expect(calendarIcon).toBeVisible();
+    await expect(calendarIcon).toHaveCSS("color", "rgba(0, 0, 0, 0.3)");
+  });
+
+  test(`should render Date with read only prop`, async ({ mount, page }) => {
+    await mount(<DateInputCustom readOnly />);
+
+    const input = getDataElementByValue(page, "input");
+    await expect(input).toHaveAttribute("readonly", /.*/);
```

Comment:
```suggestion
    await expect(input).not.toBeEditable();
```

**suggestion** - I think this is the best way in Playwright of checking something is readonly. And I particularly dislike using a regular expression to say the attribute can have absolutely any value.